docs/src/main/asciidoc/datasource.adoc

```diff
@@ -3,62 +3,61 @@ This guide is maintained in the main Quarkus repository
 
-This guide is mainly about datasource configuration.
-If you want more details about how to consume and make use of a reactive datasource,
-please refer to the xref:reactive-sql-clients.adoc[Reactive SQL clients guide].
+For more advanced configuration with examples, see xref:datasource-reference[Datasource references].
 
-== TL;DR
+[[dev-services]]
+=== Zero-config setup in development mode
 
-This is a quick introduction to datasource configuration.
-If you want a better understanding of how all this works, this guide has a lot more information in the subsequent paragraphs.
+When testing or running in development mode, Quarkus can provide you with a zero-config database, a feature referred to as _Dev Services_.
 
-[[dev-services]]
-=== Zero Config Setup (Dev Services)
+Depending on your database type, a Docker or Podman container runtime needs to be installed to use this feature.
+Some databases, for example H2, run in memory and do not require a container runtime.
 
-When testing or running in dev mode Quarkus can even provide you with a zero config database out of the box, a feature
-we refer to as Dev Services. Depending on your database type you may need Docker installed in order to use this feature.
+To start using Dev Services, include the relevant extension for the type of database you want, such as `jdbc-postgresql`, `reactive-pg-client`, or both.
 
-If you want to use Dev Services then all you need to do is include the relevant extension for the type of database you want,
-e.g. `jdbc-postgresql`, `reactive-pg-client` or both. Don't configure a database URL, username and password -
-Quarkus will provide the database and you can just start coding without worrying about config.
+In this scenario, configuring a database URL and setting up a username and password is not needed.
+If you provide the user credentials, Quarkus will configure the underlying database to use them.
+This is useful when you want to connect to the underlying database externally using your tooling.
 
-See xref:databases-dev-services.adoc[Databases Dev Services] for more details and optional configurations.
+Quarkus provides the database instance and the necessary connection between the database and the application. 
+You can start coding without worrying about configuration.
 
-=== JDBC datasource
+For more details and optional configurations, see xref:databases-dev-services.adoc[Databases Dev Services].
 
-Add the `agroal` extension plus one of `jdbc-db2`, `jdbc-derby`, `jdbc-h2`, `jdbc-mariadb`, `jdbc-mssql`, `jdbc-mysql`, `jdbc-oracle` or `jdbc-postgresql`.
+=== Configure a JDBC datasource
 
-Then configure your datasource:
+. Add the `agroal` extension plus one of `jdbc-db2`, `jdbc-derby`, `jdbc-h2`, `jdbc-mariadb`, `jdbc-mssql`, `jdbc-mysql`, `jdbc-oracle` or `jdbc-postgresql`.
```

Comment:
1. It's not actually required to add the "agroal extension", we pull it in automatically.
2. Shouldn't we describe also what needs to be done when one wants to use a JDBC driver which isn't in this list?

Reply:
There's a section below dedicated to using unlisted JDBC drivers, so I'd keep this one short. This section is meant to be a "quick reference" to get you started in most common cases.
So, no need to mention Agroal in this section either, IMO.